doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+    sample_period = conf->sample_period;
+
+    return KAA_ERR_NONE;
+}
+
+int main(void)
+{
+    /* Init random generator used to generate temperature */
+    srand(time(NULL));
+
+    /* Prepare Kaa client. */
+
+    kaa_client_t *kaa_client = NULL;
+    kaa_error_t error = kaa_client_create(&kaa_client, NULL);
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Configure notification manager. */
+
+    kaa_configuration_root_receiver_t receiver = {
+        .context = NULL,
+        .on_configuration_updated = on_configuration_updated
+    };
+
+    error = kaa_configuration_manager_set_root_receiver(
+        kaa_client_get_context(kaa_client)->configuration_manager,
+        &receiver);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Obtain default configuration, shipped within SDK. */
+
+    const kaa_root_configuration_t *dflt = kaa_configuration_manager_get_configuration(
+        kaa_client_get_context(kaa_client)->configuration_manager);
+
+    printf("Default sample period: %i seconds\n", dflt->sample_period);
+
+    /* Configure data collection. */
+
+    void *log_storage_context         = NULL;
+    void *log_upload_strategy_context = NULL;
+
+    /* The internal memory log storage distributed with Kaa SDK. */
+    error = ext_unlimited_log_storage_create(&log_storage_context,
+        kaa_client_get_context(kaa_client)->logger);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Create a strategy based on timeout. */
+    error = ext_log_upload_strategy_create(
+        kaa_client_get_context(kaa_client), &log_upload_strategy_context,
+        KAA_LOG_UPLOAD_BY_TIMEOUT_STRATEGY);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Strategy will upload logs every 5 seconds. */
+    error = ext_log_upload_strategy_set_upload_timeout(log_upload_strategy_context, 5);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Specify log bucket size constraints. */
+    kaa_log_bucket_constraints_t bucket_sizes = {
+         .max_bucket_size       = 32,   /* Bucket size in bytes. */
+         .max_bucket_log_count  = 2,    /* Maximum log count in one bucket. */
+    };
+
+    /* Initialize the log storage and strategy (by default it is not set). */
+    error = kaa_logging_init(kaa_client_get_context(kaa_client)->log_collector,
+        log_storage_context, log_upload_strategy_context, &bucket_sizes);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    /* Start Kaa SDK's main loop. example_callback will be called 1 time per second. */
+
+    error = kaa_client_start(kaa_client, example_callback, kaa_client, 1);
+
+    /* Should get here only after Kaa stop. */
+
+    kaa_client_destroy(kaa_client);
+
+    if (error) {
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+```
+
+</div>
+
+<div id="app-java" class="tab-pane fade" markdown="1" >
+
+```java
+java java java
```

Comment:
Again, please remove the stub. Rather, add a task for this part of documentation.